migrations/0035-add-handle-column-to-users-table.ts

```diff
@@ -0,0 +1,36 @@
+import { MigrationInterface, QueryRunner, TableColumn, TableIndex } from 'typeorm';
+
+const TABLE_NAME = 'Users';
+
+export class AddHandleColumnToUsersTable1667512340166 implements MigrationInterface {
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    // Add "handle" column to "Users" table as nullable initially since it does not contain any data
+    await queryRunner.addColumn(TABLE_NAME, new TableColumn({
+      name: 'handle',
+      type: 'varchar(255)',
+      isNullable: true,
+      isUnique: true,
+    }));
+
+    await queryRunner.createIndex(TABLE_NAME, new TableIndex({
+      name: 'users_by_handle_index',
+      columnNames: ['handle'],
+      isUnique: true,
+    }));
+
+    // Populate the "handle" column of "Users" table in every row with a default value
+    // formatted as {firstName}-{lastName}-{hash} truncated to 32 characters
+    await queryRunner.query(`
+      UPDATE "${TABLE_NAME}"
+      SET handle = lower(substr("firstName" || '-' || "lastName", 0, 26) || '-' || substr(md5(random()::text), 0, 7))
+    `);
```

Comment:
Is this setting the same length hash as the one that's generated on user signup? It looks like this one is of length 7, but the signup one is of length 6:
```ts
// Hexadecimals look like 0x1b9Dle so we have to truncate the fixed '0x'.
const hashValue = faker.datatype.hexaDecimal(6).slice(2);
```

Reply:
Tested the migration again to confirm that the math is right - not sure why the substr arguments are so weird but it seems like its not inclusive for the start or end index so length is 25 (handle) + 5 hash + 2 dashes = 32

![Screenshot 2023-03-29 at 6 31 26 PM](https://user-images.githubusercontent.com/33165426/228704887-cc6669ff-01da-426e-a2d0-2335b7fa35fc.jpg)

Reply:
>not sure why the substr arguments are so weird but it seems like its not inclusive for the start or end index so length is 25 (handle) + 5 hash + 2 dashes = 32

It seems like the hashes here are of length 6 though? Looks like the handle including the first '-' is of length 25, the next dash is 1, and the hash is 6, resulting in 32 total. The reason why the start/end positioning might be weird is because I think Postgres does 1-based indexing (see example [here](https://www.postgresqltutorial.com/postgresql-string-functions/postgresql-substring/)), so starting at 0 may make it include an empty string at the start which includes an extra character. Otherwise, I'm unsure why specifying `26` makes the string length 25, even if the string is longer than 26 characters.

I hate to stall this PR even further, but can we quickly replace the 0-indexing with 1-indexing and see if this results in the same behavior? I.e. let's just replace the 0, 26 with 1, 25 and the 0, 7 with 1, 6 and verify the output. I think this will sharpen our understanding of Postgres and substring indexing, while also making the code less confusing. Appreciate the patience here

Reply:
Will approve once this is verified.